### PR TITLE
[stable/3.0] Disable apache2 service when deployed with HA

### PR DESCRIPTION
**Pull request by jsuchome**
This is an attempt to fix https://bugzilla.suse.com/show_bug.cgi?id=1021106. Comments are welcomed.

### CODE REVIEW

**Comment by jsuchome** [comment]
[apache service not configured correctly with HA (not handled by pacemaker)](https://trello.com/c/1b5iNWF7/273-apache-service-not-configured-correctly-with-ha-not-handled-by-pacemaker)

File: chef/cookbooks/crowbar-pacemaker/recipes/apache.rb [hunk 1]
```diff
@@ -101,3 +101,8 @@
 # Override service provider for apache2 resource defined in apache2 cookbook
 resource = resources(service: "apache2")
 resource.provider(Chef::Provider::CrowbarPacemakerService)
+
+# Disable apache2 as it should be only managed by pacemaker now
+service "apache2" do
+  action :disable
+end
```
Comment (vuntz) [inline chef/cookbooks/crowbar-pacemaker/recipes/apache.rb]: This is unfortunately too fragile, as something could come later on in the run-list that would re-enable it :/

**Comment by vuntz** [comment]
That being said, we have this code already:

```ruby
# Override service provider for apache2 resource defined in apache2 cookbook
resource = resources(service: "apache2")
resource.provider(Chef::Provider::CrowbarPacemakerService)
```
This should do all the magic that is required for us, as any call to `:enable` will effectively disable the service (because that's what the provider is doing).

So something may not be working here?

**Comment by jsuchome** [comment]
``resource = resources(service: "apache2")``

I think the point is that it was already enabled before this code

File: chef/cookbooks/crowbar-pacemaker/recipes/apache.rb (hunk 1 above)
Comment (jsuchome) [inline chef/cookbooks/crowbar-pacemaker/recipes/apache.rb]: But isn't this what we're doing for drbd?

**Comment by vuntz** [comment]
> I think the point is that it was already enabled before this code

In the compile phase? (could be, in which case we need the attribute like in your other PR)

**Comment by vuntz** [comment]
I believe this is the right fix: https://github.com/crowbar/crowbar-core/pull/1018 (works for me)

**Pull request closed without merging**